ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -1197,14 +1213,49 @@ reg_t registers[REG_COUNT];
 void setCurrBSDFParams(in vec3 n, in vec3 L)
 {
 	vec3 campos = irr_glsl_SBasicViewParameters_GetEyePos(CamData.params.NormalMatAndEyePos);
-	irr_glsl_ViewSurfaceInteraction interaction = irr_glsl_calcFragmentShaderSurfaceInteraction(campos, WorldPos, n);
+	irr_glsl_IsotropicViewSurfaceInteraction interaction = irr_glsl_calcFragmentShaderSurfaceInteraction(campos, WorldPos, n);
 	irr_glsl_BSDFIsotropicParams isoparams = irr_glsl_calcBSDFIsotropicParams(interaction, L);
 	//TODO: T,B tangents
 	vec3 T = vec3(1.0,0.0,0.0);
 	vec3 B = vec3(0.0,0.0,1.0);
 	currBSDFParams = irr_glsl_calcBSDFAnisotropicParams(isoparams, T, B);
```

Comment:
T can be extracted from your bump-mapping code (kind-of), and use the `irr_glsl_calcBSDFAnisotropicParams` overload that computes `B` from a `cross` otherwise sampling will fall apart bigtime.

Reply:
did you make the changes?

Reply:
?

Reply:
not relevant any more